src/irmin-pack/pack_index.ml

```diff
@@ -35,9 +35,9 @@ module Make (K : Irmin.Hash.S) = struct
   module Key = struct
     type t = K.t [@@deriving irmin]
 
-    let hash = Irmin.Type.(unstage (short_hash K.t)) ?seed:None
+    let hash t = Irmin.Type.(unstage (short_hash K.t)) ?seed:None t |> abs
```

Comment:
can you replace `abs` by a bitwise operation?

Reply:
Done